### ISSUE TITLE
Error handling: program temp_pystata_kernel_program_name already defined

### ISSUE THREAD

**Issue by gaksaray** [ticoneva/pystata-kernel#24]
Running a cell with an error twice throws the error instead of the original error:
```
program temp_pystata_kernel_program_name already defined
r(110);
```

To demonstrate, say I want to create a table:

<img width="628" alt="Screen Shot 2023-03-24 at 10 08 23 PM" src="https://user-images.githubusercontent.com/72267919/227653089-f3ca69f1-7828-4890-945f-2ce0091c3b96.png">

Here I intentionally specified `delimiter` option incorrectly, which produces the error as expected. Then I ran the cell again. This time it produces a different (incomprehensible to end user) error:

<img width="646" alt="Screen Shot 2023-03-24 at 10 08 35 PM" src="https://user-images.githubusercontent.com/72267919/227653412-64970c81-1709-4bd2-9a3f-c742127e121d.png">

In fact, if I force delete that "program", it reverts back to the original Stata error:

<img width="620" alt="Screen Shot 2023-03-24 at 10 09 14 PM" src="https://user-images.githubusercontent.com/72267919/227653592-08ee99e2-1b34-4dd2-9d52-587c966f05a3.png">

Expected behavior is that it produces the same error `r(198)` as before.

**Comment by hugetim** [comment]
For "echo = None" mode, `nbstata` is more robust in this and other ways. Compare:
https://github.com/ticoneva/pystata-kernel/blob/main/pystata-kernel/helpers.py#L158
and
https://github.com/hugetim/nbstata/blob/master/nbstata/stata_more.py#L90

**Comment by ticoneva** [comment]
Fixed in the version 0.3.1.
